packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -105,7 +109,14 @@ export const AuthModal = ({
 
         const handlePasswordlessLogin = async (email) => {
             try {
-                await authorizePasswordlessLogin.mutateAsync({userid: email})
+                const callbackURL = isAbsoluteURL(passwordlessConfigCallback)
+                    ? passwordlessConfigCallback
+                    : `${appOrigin}${passwordlessConfigCallback}`
+                const redirectPath = window.location.pathname
```

Comment:
We might also need to include the query params of the current url so that the redirect also works if the user is in product search!
```suggestion
                const redirectPath = window.location.pathname + window.location.search
```

For example if they are logging in from: https://wasatch-mrt-passwordless-poc.mrt-storefront-staging.com/global/en-GB/search?q=glove